### PR TITLE
feat: `vm.publicKeyP256`

### PR DESCRIPTION
**Pull request by klkvr**
## Motivation

Closes https://github.com/foundry-rs/foundry/issues/7908

Adds cheatcode for deriving public key for P256 private key

### CODE REVIEW

File: crates/cheatcodes/spec/src/vm.rs [hunk 1]
```diff
@@ -2195,6 +2195,10 @@ interface Vm {
     #[cheatcode(group = Crypto)]
     function signP256(uint256 privateKey, bytes32 digest) external pure returns (bytes32 r, bytes32 s);
 
+    /// Signs `digest` with `privateKey` using the secp256r1 curve.
```
Comment (DaniPopes) [inline crates/cheatcodes/spec/src/vm.rs:2198]: wrong doc

File: crates/cheatcodes/src/crypto.rs [hunk 2]
```diff
@@ -230,30 +241,33 @@ fn sign_with_wallet(
...
+    let signing_key = parse_private_key_p256(private_key)?;
     let signature: Signature = signing_key.sign_prehash(digest.as_slice())?;
     let r_bytes: [u8; 32] = signature.r().to_bytes().into();
     let s_bytes: [u8; 32] = signature.s().to_bytes().into();
 
     Ok((r_bytes, s_bytes).abi_encode())
 }
 
-fn parse_private_key(private_key: &U256) -> Result<SigningKey> {
+fn validate_private_key<C: ecdsa::PrimeCurve>(private_key: &U256) -> Result<()> {
```
Comment (DaniPopes) [inline crates/cheatcodes/src/crypto.rs:252]: we could pass either C by value to use Debug or a &str parameter for the name of the curve, to use in the error message

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Review by mattsse — COMMENTED** [review]
lgtm, pending doc nit

**Review by DaniPopes — APPROVED** [review]

**Comment by pcaversaccio** [comment]
Can you please add the function to the `vm` interface in `forge-std`, otherwise hard to use 🙃

**Pull request merged**